### PR TITLE
[REMS-323] Patient info lookup clean up on PIMS

### PR DESCRIPTION
**Pull request by plarocque4**
## Describe your changes

REMS-323: Patient info lookup clean up on PIMS.
Change the required parameters to the doctorOrders getRx interface on PIMS.
Allow any value in the doctorOrder to be a query parameter as well.

Test with corresponding changes in DTR.

## Issue ticket number and Jira link

[REMS-323](https://jira.mitre.org/browse/REMS-323)

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] Ensure the target / base branch for any feature PR is set to `dev` not main (the only exception to this is releases from `dev` and hotfix branches)

## Checklist for conducting a review
- [ ] Review the code changes and make sure they all make sense and are necessary. 
- [ ] Pull the PR branch locally and test by running through workflow and making sure everything works as it is supposed to. 

## Workflow 

Owner of the Pull Request will be responsible for merge after all requirements are met, including approval from at least one reviewer. Additional changes made after a review will dismiss any approvals and require re-review of the additional updates. Auto merging can be enabled below if additional changes are likely not to be needed. The bot will auto assign reviewers to your Pull Request for you.

### CODE REVIEW

File: src/backend/routes/doctorOrders.js
Comment (rdingwell) [inline src/backend/routes/doctorOrders.js:1]: I know that this is an update to this file but is there a reason it is not a typescript file?

File: src/backend/routes/doctorOrders.js [hunk 1]
```diff
@@ -146,7 +170,13 @@ function parseNCPDPScript(newRx) {
       newRx.Message.Body.NewRx.Patient.HumanPatient.Name.FirstName +
       ' ' +
       newRx.Message.Body.NewRx.Patient.HumanPatient.Name.LastName,
+    patientFirstName: newRx.Message.Body.NewRx.Patient.HumanPatient.Name.FirstName,
```
Comment (rdingwell) [inline src/backend/routes/doctorOrders.js:173]: There is a lot of chaining of property calls here, can this be cleaned up so there are higher level variables that have been null checked.  So something like (if this was type script) 

let patient = newRx?.Message?.Body?.NewRx?.Patient?.HumanPatient; 
let name = patient.Name 
let address = patient.Address. 

Same of Prescriber.

Comment (smalho01) [reply src/backend/routes/doctorOrders.js:173]: @smalho01 address this comment while pats out

File: src/backend/routes/doctorOrders.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ router.post('/api/addRx', async (req, res) => {
   try {
     await newOrder.save(); //updating the object or adding to it
   } catch (error) {
-    console.log('ERROR! douplicate found, prescription already exists already exists');
+    console.log('ERROR! douplicate found, prescription already exists');
```
Comment (smalho01) [inline src/backend/routes/doctorOrders.js:43]: typo "duplicate"

**Review by smalho01 — APPROVED** [review]

**Pull request merged**